Adding component for installing Caddy

blarghmatey opened the pull request:
Caddy is the web server that we are using for new installations due to its ease of deployment and flexible usage patterns.

blarghmatey committed:
Adding component for managing Caddy installation

blarghmatey committed:
Added default template Caddyfile

shaidar commented on src/bilder/components/caddy/packer.pkr.hcl:
```diff
@@ -0,0 +1,94 @@
+  default = "operations-qa"
+}
+
+# Available options are "web" or "worker". Used to determine which type of node to build an image for.
```
Why is this needed here?

shaidar commented on src/bilder/components/caddy/packer.pkr.hcl:
```diff
@@ -0,0 +1,94 @@
+    ]
+  }
+  provisioner "shell-local" {
+    except = ["docker.concourse"]
```
Why this exception?

blarghmatey replied:
Copy :spaghetti:

shaidar replied:
I thought this was copy 🍝

shaidar commented on src/bilder/components/caddy/packer.pkr.hcl:
```diff
@@ -0,0 +1,94 @@
+  # Base all builds off of the most recent Debian 10 image built by the Debian organization.
+  source_ami_filter {
+    filters = {
+      name                = "debian-10-amd64*"
```
Should this be defined in the model instead?

blarghmatey replied:
Ideally the config should be agnostic to what it's actually being launched on so I don't think it makes sense to mix those concerns.

shaidar commented on src/bilder/components/caddy/steps.py:
```diff
@@ -0,0 +1,116 @@
+            dest="/usr/local/bin/caddy",
+            src=caddy_config.custom_download_url(),
+            mode=755,
+            sttate=state,
```
Typo

blarghmatey committed:
Added management of SystemD unit for Caddy installed without package manager

blarghmatey committed:
Adding tests, fixing Packer and updating component templates

- Added sample_deploy and tests for Caddy build
- Fixed up Packer config
- Updated template for components to match project changes

shaidar commented:
👍
